src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -773,13 +773,13 @@ private static JsonRpcMessage CreateError(JToken id, Exception exception)
                 foreach (var targetMap in this.targetRequestMethodToClrMethodMap)
                 {
                     targetMethod = new TargetMethod(request, targetMap.Item1, jsonSerializer, targetMap.Item2);
-                    if (targetMethod.IsFound)
+                    if (targetMethod != null && targetMethod.IsFound)
                     {
                         break;
                     }
                 }
 
-                if (!targetMethod.IsFound)
+                if (targetMethod == null || !targetMethod.IsFound)
```

Comment:
I actually don't think targetMethod can ever be null.  The only scenario where that's possible is if targetRequestMethodToClrMethodMap is empty.  But we already check for that beforehand and return a different error.  In any case, I added a test to test that a remote target exception is thrown if we call InvokeAsync with no target object.